hatch_build.py

```diff
@@ -0,0 +1,31 @@
+from pathlib import Path
+from sysconfig import get_platform
+from typing import Any
+
+from hatch.utils.platform import Platform
+from hatchling.builders.hooks.plugin.interface import BuildHookInterface
+
+
+class CustomBuildHook(BuildHookInterface):
+    def clean(self, versions: list[str]):
+        build_dir = Path(self.directory)
+        parser_bin = build_dir / "parser"
+
+        if parser_bin.is_file():
+            parser_bin.unlink()
+
+        return super().clean(versions)
+
+    def initialize(self, version: str, build_data: dict[str, Any]):
+        self.app.display_waiting("Bootstrapping parser")
+
+        Platform().check_command(["./bootstrap.sh"], shell=False)
+
+        self.app.display_success("Done!")
+
+        build_data["force_include"]["dist/parser"] = "glang/bin/parser"
+
+        build_data["pure_python"] = False
+        build_data["infer_tag"] = True
+
+        return super().initialize(version, build_data)
```

Comment:
Is this only run once when I create the environment? Let's say I wanted to redo the bootstrap... would I do it manually, rebuild hatch, or something else entirely?

Reply:
Yes. It's only run when you build the package (which you implicitly do the first time you create an environment). You can also run `hatch build` to manually rebuild, or just `hatch run dev:bootstrap` to only run `bootstrap.sh` in the `dev` environment (this is defined in `pyproject.toml`.